custom_components/myskoda/climate.py

```diff
@@ -123,10 +130,117 @@ async def async_set_temperature(self, **kwargs):  # noqa: D102
+        return [HVACMode.HEAT, HVACMode.OFF]
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:  # noqa: D102
+        if ac := self._air_conditioning():
+            if ac.state == AirConditioningState.HEATING_AUXILIARY:
+                return HVACMode.HEAT
+            return HVACMode.OFF
+
+    @property
+    def hvac_action(self) -> HVACAction | None:  # noqa: D102
+        if ac := self._air_conditioning():
+            if ac.state == AirConditioningState.HEATING_AUXILIARY:
+                return HVACAction.HEATING
+            return HVACAction.OFF
+
+    @property
+    def target_temperature(self) -> None | float:  # noqa: D102
+        if ac := self._air_conditioning():
+            target_temperature = ac.target_temperature
+            if target_temperature is None:
+                return
+            return target_temperature.temperature_value
+
+    @Throttle(timedelta(seconds=API_COOLDOWN_IN_SECONDS))
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode):  # noqa: D102
+        if ac := self._air_conditioning():
+            target_temperature = ac.target_temperature
+            if target_temperature is None:
+                return
+
+            if hvac_mode == HVACMode.HEAT:
+                spin = self.coordinator.config.options.get(CONF_SPIN)
+                if spin is not None:
+                    if (
+                        ac.state != AirConditioningState.OFF
+                        and ac.state != AirConditioningState.HEATING_AUXILIARY
+                    ):
+                        await self.coordinator.myskoda.stop_air_conditioning(
+                            self.vehicle.info.vin
+                        )
+                    await self.coordinator.myskoda.start_auxiliary_heating(
+                        self.vehicle.info.vin,
+                        target_temperature.temperature_value,
+                        spin,
+                    )
+                else:
+                    _LOGGER.error("Cannot start auxiliary heater: No S-PIN set.")
+            else:
+                await self.coordinator.myskoda.stop_auxiliary_heating(
+                    self.vehicle.info.vin
+                )
+            _LOGGER.info("Auxiliary HVAC mode set to %s.", hvac_mode)
+
+    async def async_turn_on(self):  # noqa: D102
+        await self.async_set_hvac_mode(HVACMode.HEAT)
+
+    async def async_turn_off(self):  # noqa: D102
+        await self.async_set_hvac_mode(HVACMode.OFF)
+
+    @Throttle(timedelta(seconds=API_COOLDOWN_IN_SECONDS))
+    async def async_set_temperature(self, **kwargs):  # noqa: D102
+        temp = kwargs[ATTR_TEMPERATURE]
+        await self.coordinator.myskoda.set_target_temperature(
+            self.vehicle.info.vin, temp
+        )
+        _LOGGER.info("Target temperature for auxiliary heater set.")
+
+    def required_capabilities(self) -> list[CapabilityId]:
+        return [
+            CapabilityId.AIR_CONDITIONING_HEATING_SOURCE_AUXILIARY,
```

Comment:
Electric source is used for air-conditioning entity, and auxiliary source for auxiliary entity ... If I start auxiliary-heater with electric type instead, it will act exactly same as when air condition is turned on so I don't see any advantage of putting this capability inside.. 

Probably we will need to fine-tune the best capability, because from fixtures it seems like only Superb iV has AIR_CONDITIONING_HEATING_SOURCE_AUXILIARY. Kodiaq and Octavia is using  AUXILIARY_HEATING but that one is not available in Superb iV..